portfolio/src/main/webapp/script.js

```diff
@@ -33,10 +33,68 @@ async function getContent() {
   document.getElementById('message-container').innerText = message;
 }
 
+function postComments(comments){
+    console.log(comments);
+    const messageBoard = document.getElementById('comments-container');
+    comments.forEach((comment) => {
+      messageBoard.appendChild(createCmtEl(comment));
+    }) 
+}
+
+function deleteAllComments(){
+    fetch('/delete-data',{method:"POST"}).then(response => response.json()).then((emptyComments) => {
+        console.log(emptyComments);
+    });
+    getComments();
+}
+
+function deleteComment(){
+    fetch('/delete-data?comment-key='+getCommentKey()).then(response => response.json()).then((emptyComments) => {
+        console.log(emptyComments);
+    });
+    getComments();
+}
+
+function getCommentKey(){
+    let searchParams = (new URL(document.location)).searchParams;
+    let userNum = searchParams.get("comment-key");
+    if(commentkey == null || commentkey.length === 0){
+        return "1";
+    }
+    return commentkey;
+}
+
 function getComments(){
-   String json = fetch('/data')
-   response.setContentType("text/html;");//going to priint directly on page
-   response.getWriter().println(json);;
-  }
+   //confirm((new URL(document.location)).searchParams);
+   fetch('/data?user-comment-num='+getUserNum()).then(response => response.json()).then((comments) => {
+    //need to have the ?user-comment-num in order to pass correct params
+    postComments(comments);    
+  });
+}
+
+function getUserNum(){
+    //the current document's URL params
+    let searchParams = (new URL(document.location)).searchParams;
+    //that returns a map like structure, so we can get out desired param with .get()
+    let userNum = searchParams.get("user-comment-num");
+    if(userNum == null || userNum.length === 0){
+        return "1";
+    }
+    return userNum;
+}
+
+function createCmtEl(comment) {
+  const comElem = document.createElement('li');
+  comElem.className = 'Comment';
+
+  const nameElem = document.createElement('span');
+  nameElem.innerText = comment.name;
+
+  const txtElem = document.createElement('span');
+  var str = comment.name +"commented: \n " + comment.text;
```

Comment:
Looks like this will print the name twice, and will not have space between name and the "commented" string.